R/wwinference.R

```diff
@@ -62,10 +62,7 @@
 #'
 wwinference <- function(ww_data,
                         count_data,
-                        model_spec = wwinference::get_model_spec(
-                          forecast_date =
-                            "2023-12-06"
-                        ),
+                        model_spec = wwinference::get_model_spec(),
```

Comment:
Perhaps I am missing something, but I do not think `model_spec` should have a default value. I don't see why `wwinference` should expect a particular `model_spec` by default any more than it expects particular `ww_data` or `count_data` values. 

Not for this PR, but I would in the long run suggest doing away with the `model_spec` construct. For example, I don't think the generation interval (as a vector) and the path to a parameter file (as a strong) should be passed to `wwinference` as part of the same list, while the `ww_data` and `count_data` are passed separately.

Reply:
Note that documentation will need to change too.

Reply:
Hmmm I am not sure I am completely following -- we had talked a while back about setting this up this way, similar to how `epinowcast`/`epinow2()` takes in a function that generates a list with defaults, so that instead of writing all the separate inputs (which we used to do in the other fitting functions in the `cfaforecastrenewalww` package, this just calls a function with defaults to generate those). My understanding was that this was better practice, especially since we are setting up this package to by default fit covid data and genome concentration data and all of the parameters/delay distributions being specified are specific to that.

https://github.com/epinowcast/epinowcast/blob/5aa7bcfae0766e14938ff1ad2c9c2f649ed7e86c/R/epinowcast.R#L106

Maybe I misunderstood that conversation but this was my attempt at implementing that. Tagging @seabbs and @gvegayon bc I think they were part of that convo

Reply:
I don't recall the conversation, but I am not against having defaults like this. This makes things easier when starting with the package and makes it even more appealing to new users. Nonetheless, perhaps it would be a good idea to (in a separate PR) add good **`print` and `summary` methods that explicitly clarify the model specs**.

An argument in favor of having defaults like this is that you have spent no trivial amount of time thinking about these numbers, so having a well-thought starting point is more beneficial than anything else to users.

Reply:
Yeah I think that is a good idea for #59